### PR TITLE
Add `esphome.subdomain.conf.sample`

### PR DESCRIPTION
**Pull request by golles**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/reverse-proxy-confs/blob/master/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description
Subdomain configuration for [ESPHome](https://esphome.io)

## Benefits of this PR and context

This is an addition to your amazing collection of configurations. I wanted to share this one

## How Has This Been Tested?
I've been using this config since Oct 7, 2020.

## Source / References

https://esphome.io

### CODE REVIEW

**Comment by golles** [comment]
Feel free to decline this PR if you think it has no added value.

File: esphome.subdomain.conf.sample [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    include /config/nginx/ssl.conf;
+
+    client_max_body_size 0;
+
+    # enable for ldap auth (requires ldap-location.conf in the location block)
+    #include /config/nginx/ldap-server.conf;
+
+    # enable for Authelia (requires authelia-location.conf in the location block)
+    include /config/nginx/authelia-server.conf;
```
Comment (nemchik) [inline esphome.subdomain.conf.sample:19]:
```suggestion
    #include /config/nginx/authelia-server.conf;
```

**Review by nemchik — CHANGES_REQUESTED** [review]
Auth lines need to be commented by default. Otherwise it looks good to me.

File: esphome.subdomain.conf.sample (hunk 1 above)
Comment (golles) [inline esphome.subdomain.conf.sample:19]:
```suggestion
    #include /config/nginx/authelia-server.conf;
```

**Comment by golles** [comment]
I've created #554 to replace this PR, sorry for the trouble

**Pull request closed without merging**